math/src/test/kotlin/ktx/math/ImmutableVector2Test.kt

```diff
@@ -0,0 +1,1436 @@
+
+    @Test
+    fun `times should return same result than Vector2 scale`() {
+        scalars.forEach { factor ->
+            vectors.forEach { vector ->
+                assertEquals(vector.toMutable().scl(factor).toImmutable(), vector * factor)
+            }
+        }
+    }
+
+    @Test
+    fun `times should return scaled vector`() {
+        assertEquals(ImmutableVector2.ZERO, ImmutableVector2.ZERO * 42f)
+        assertEquals(ImmutableVector2(0f, 2f), ImmutableVector2.Y * 2f)
+        assertEquals(ImmutableVector2(15f, -6f), ImmutableVector2(5f, -2f) * 3f)
+        assertEquals(ImmutableVector2(2.5f, -1f), ImmutableVector2(5f, -2f) * 0.5f)
+    }
+
+    @Test
+    fun `times should return same result than scl does for Vector2`() {
+        vectors.forEach { v1 ->
+            vectors.forEach { v2 ->
+                assertEquals(v1.toMutable().scl(v2.toMutable()).toImmutable(), v1 * v2)
+                assertEquals(v1.toMutable().scl(v2.x, v2.y).toImmutable(), v1.times(v2.x, v2.y))
+            }
+        }
+    }
+
+    @Test
+    fun `times should return scaled vector on both axis`() {
+        assertEquals(ImmutableVector2.ZERO, ImmutableVector2.X * ImmutableVector2.ZERO)
+        assertEquals(ImmutableVector2.ZERO, ImmutableVector2.X.times(0f, 0f))
+
+        assertEquals(ImmutableVector2.ZERO, ImmutableVector2.X * ImmutableVector2.Y)
+        assertEquals(ImmutableVector2.ZERO, ImmutableVector2.X.times(0f, 1f))
+
+        assertEquals(ImmutableVector2(4f, -15f), ImmutableVector2(-2f, 3f) * ImmutableVector2(-2f, -5f))
+        assertEquals(ImmutableVector2(4f, -15f), ImmutableVector2(-2f, 3f).times(-2f, -5f))
+
+        assertEquals(ImmutableVector2(4f, 15f), ImmutableVector2(2f, 3f) * ImmutableVector2(2f, 5f))
+        assertEquals(ImmutableVector2(4f, 15f), ImmutableVector2(2f, 3f).times(2f, 5f))
+    }
+
+    @Test
+    fun `dot should return same result than Vector2`() {
+        vectors.forEach { v1 ->
+            vectors.forEach { v2 ->
+                assertEquals(v1.toMutable().dot(v2.toMutable()), v1 dot v2)
+                assertEquals(v1.toMutable().dot(v2.x, v2.y), v1.dot(v2.x, v2.y))
+            }
+        }
+    }
+
+    @Test
+    fun `dot should return dot product`() {
+        assertEquals(1f, ImmutableVector2.X dot ImmutableVector2.X)
+        assertEquals(1f, ImmutableVector2.X dot ImmutableVector2.X)
+
+        assertEquals(1f, ImmutableVector2.Y.dot(0f, 1f))
+        assertEquals(1f, ImmutableVector2.Y.dot(0f, 1f))
+
+        assertEquals(0f, ImmutableVector2.X.dot(0f, 1f))
+        assertEquals(0f, ImmutableVector2.X.dot(0f, 1f))
+
+        assertEquals(0f, ImmutableVector2.X.dot(0f, 0f))
+        assertEquals(0f, ImmutableVector2.X.dot(0f, 0f))
+
+        assertEquals(2f, ImmutableVector2(1f, 1f).dot(1f, 1f))
+        assertEquals(2f, ImmutableVector2(1f, 1f).dot(1f, 1f))
+
+        assertEquals(7f, ImmutableVector2(-2f, 3f).dot(4f, 5f))
+        assertEquals(7f, ImmutableVector2(-2f, 3f).dot(4f, 5f))
```

Comment:
Why are you invoking these methods twice with the same parameters? Did you want to compare it to the mutable vector results or test both dot functions (vector/2 floats)?